src/Umbraco.Core/Composing/ComponentCollection.cs

```diff
@@ -43,8 +44,15 @@ public void Terminate()
                     var componentType = component.GetType();
                     using (_logger.DebugDuration<ComponentCollection>($"Terminating {componentType.FullName}.", $"Terminated {componentType.FullName}.", thresholdMilliseconds: LogThresholdMilliseconds))
                     {
-                        component.Terminate();
-                        component.DisposeIfDisposable();
+                        try
+                        {
+                            component.Terminate();
+                            component.DisposeIfDisposable();
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.Error(componentType, ex, "Error while terminating component.");
```

Comment:
```suggestion
                            _logger.Error(componentType, ex, $"Error while terminating component '{componentType.FullName}'.");
```

Reply:
Just a super minor thing to help with readability of logs that the rendered message log contains the class/type of the component. Yes, it already exists as the `SourceContext` on the log, but this makes it easier to read than just `Error while terminating component.`

Reply:
Was hoping to update your PR directly @ronaldbarendse (I don't have permissions to push/update your PR)
I will merge this PR but will amend with the minor change I have suggested

Reply:
That probably makes it easier to search indeed, although I would then change the source context to ‘ComponentCollection‘, as that is the calling type and the same as the other logging calls...

Reply:
@warrenbuckley Can we change this minor thing before it gets released? As the error itself isn't logged from the component and all other logging within this file use `ComponentCollection` as source context, it's best to change this into:
```suggestion
                            _logger.Error<ComponentCollection>(ex, "Error while terminating component {ComponentType}.", componentType.FullName);
```

Reply:
Good spot @ronaldbarendse I have fixed in this commit here - d378495942fe1828571444aeb45e84143059f3c4